src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,57 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include <yaml-cpp/yaml.h>
+#include "rgw_tracer.h"
+#include "common/debug.h"
+#include "global/global_context.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+thread_local std::unique_ptr<tracing::Tracer> rgw_tracer = std::unique_ptr<tracing::Tracer>(new tracing::Tracer(jaeger_configuration::jaeger_default_config));
+
+namespace tracing {
+
+const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+
+Tracer::Tracer(jaegertracing::Config& conf):open_tracer(jaegertracing::Tracer::make(conf)) {};
+
+Tracer::Tracer(const std::string& config_file_path) {
+  const auto config_yaml = YAML::LoadFile(config_file_path);
+  const auto config = jaegertracing::Config::parse(config_yaml);
+  open_tracer =  jaegertracing::Tracer::make(config);
+};
+
+std::unique_ptr<opentracing::Span> Tracer::start_trace(const std::string& trace_name) {
+  if(is_enabled()) {
+    return open_tracer->StartSpan(trace_name);
+  }
+  return noop_tracer->StartSpan(trace_name);
+}
+
+std::unique_ptr<opentracing::Span> Tracer::start_span(const std::string& span_name, std::unique_ptr<opentracing::Span>& parent_span) {
+  if(is_enabled()) {
+    return open_tracer->StartSpan(span_name, {opentracing::ChildOf(&parent_span->context())});
+  }
+  return noop_tracer->StartSpan(span_name);
```

Comment:
the `opentracing::Tracer` api for `StartSpan()` takes the first argument as a `opentracing::string_view`, so these wrappers should use that type instead of `const std::string&`

most callers will pass a string literal here, like:
```c++
  rgw_tracer->start_span("verify_permission", this->trace);
```
converting this `const char*` into a `std::string` can be expensive - it has to copy the whole string, and potentially allocate memory if the string is long enough. `string_view` doesn't allocate or copy